### PR TITLE
Fix stale xds cache when discoveryRequest requests xds happend before…

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
… proxy sidecarscope updated

**Please provide a description of this PR:**

https://github.com/istio/istio/issues/34549#issuecomment-895947974

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by ramaraochavali — COMMENTED** [review]
looks like a good approach. Can you add some comments?

**Comment by howardjohn** [comment]
LGTM

/retest

File: pilot/pkg/xds/ads.go [hunk 1]
```diff
@@ -234,6 +234,12 @@ func (s *DiscoveryServer) processRequest(req *discovery.DiscoveryRequest, con *C
 
 	request.Reason = append(request.Reason, model.ProxyRequest)
 	request.Start = time.Now()
+	// SidecarScope for the proxy may has not been updated based on this pushContext.
```
Comment (ramaraochavali) [inline pilot/pkg/xds/ads.go:237]: /proxy may has not been updated -> proxy may not have been updated

Comment (ramaraochavali) [reply pilot/pkg/xds/ads.go:237]: thanks for comment btw, it is really useful

Comment (hzxuzhonghu) [reply pilot/pkg/xds/ads.go]: Done

**Review by ramaraochavali — APPROVED** [review]

**Pull request merged**